.github/workflows/integration_test_run.yaml

```diff
@@ -196,6 +196,33 @@ jobs:
           microk8s-addons: ${{ inputs.microk8s-addons }}
           channel: ${{ inputs.channel }}
           juju-channel: ${{ inputs.juju-channel }}
+      - name: Generate charmcraft cache key
+        run: |
+          CHARM_NAME=$(yq .name ./metadata.yaml)
+          CHARMCRAFT_BASE="$(yq -o=j -I=0 '.bases | sort_keys(..)' ./charmcraft.yaml)"
+          CHARMCRAFT_BASE=$(printf %s "$CHARMCRAFT_BASE"|jq -sRr @uri)
+          CHARMCRAFT_CACHE_KEY_BASE="charmcraft-cache?name=$CHARM_NAME&bases=$CHARMCRAFT_BASE"
+          echo "CHARMCRAFT_CACHE_KEY=$CHARMCRAFT_CACHE_KEY_BASE&date=$(date +%Y-%m-%d)" >> $GITHUB_ENV
+          echo 'CHARMCRAFT_CACHE_ALT_KEYS<<EOF' >> $GITHUB_ENV 
+          for d in {1..2}
+            do echo "$CHARMCRAFT_CACHE_KEY_BASE&date=$(date -d"-$d days" +%Y-%m-%d)" >> $GITHUB_ENV
+          done
+          echo 'EOF' >> $GITHUB_ENV
+      - name: Restore charmcraft container cache
```

Comment:
Out of curiosity: Would be possible to have a situation where we might prefer to not use the cache (charmcraft and rockcraft images)?

Reply:
The normal integration test workflows only consume caches, there will be two more workflows that build caches. So if certain charm doesn't want to use caches, it can just simply not setup the cache building workflows.